### PR TITLE
feat(logbook): add logbook integration 

### PR DESCRIPTION
**Pull request by zarirhamza**
Adds logging support for`logbook` as part of a larger initiative to improve APM logging capabilities.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Comment by zarirhamza** [comment]
This can only be merged after deployment of this [PR](https://github.com/DataDog/logs-backend/pull/59893) scheduled by EOD November 13

File: ddtrace/contrib/logbook/patch.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    dict(),
+)
+
+
+def get_version():
+    # type: () -> str
+    return getattr(logbook, "__version__", "")
+
+
+def _tracer_injection(event_dict):
```
Comment (mabdinur) [inline ddtrace/contrib/logbook/patch.py:29]: You can do this as a follow up: We can refactor this code. It is duplicated in our other logging integrations

File: ddtrace/contrib/logbook/__init__.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+The trace values are patched to every log at the top level of the record. In order to correlate
+logs, it is highly recommended to use JSON logs which can be achieved by using a handler with
+a proper formatting::
+
+    handler = FileHandler(format_string='{{\"message\": "{record.message}", '
+                                        '\"dd.trace_id\": "{record.dd.trace_id}", '
+                                        '\"dd.span_id\": "{record.dd.span_id}", '
+                                        '\"dd.env\": "{record.dd.env}", '
+                                        '\"dd.service\": "{record.dd.service}", '
+                                        '\"dd.version\": "{record.dd.version}"}}')
```
Comment (mabdinur) [inline ddtrace/contrib/logbook/__init__.py:38]: We should update this to reflect the new format

File: ddtrace/contrib/logbook/patch.py (hunk 1 above)
Comment (zarirhamza) [reply ddtrace/contrib/logbook/patch.py:29]: Noted

File: ddtrace/contrib/logbook/__init__.py (hunk 2 above)
Comment (zarirhamza) [reply ddtrace/contrib/logbook/__init__.py]: Done

**Review by mabdinur — APPROVED** [review]

**Pull request merged**